lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```diff
@@ -151,7 +151,11 @@ public function lastInsertId($name = null)
             return false;
         }
 
-        OraclePlatform::assertValidIdentifier($name);
+        $identifiers = explode('.', $name);
+
+        foreach ($identifiers as $identifier) {
```

Comment:
> Why not let the DB fail invalid identifiers.

Probably OK, yes. Our job is just verifying that proper quoting is used then.